Lib/webbrowser.py

```diff
@@ -599,6 +602,82 @@ def open(self, url, new=0, autoraise=True):
             rc = osapipe.close()
             return not rc
 
+#
+# Platform support for iOS
+#
+if sys.platform == "ios":
+    try:
+        from ctypes import cdll, c_void_p, c_char_p, c_ulong, util
+    except ImportError:
+        # If ctypes isn't available, we can't trigger the browser
+        objc = None
+    else:
+        # ctypes is available. Load the ObjC library, and wrap the
+        # objc_getClass, sel_registerName and objc_msgSend methods
+        objc = cdll.LoadLibrary(util.find_library("objc"))
+        if objc._name:
+            objc.objc_getClass.restype = c_void_p
+            objc.objc_getClass.argtypes = [c_char_p]
+            objc.sel_registerName.restype = c_void_p
+            objc.sel_registerName.argtypes = [c_char_p]
+            # The return type of objc_msgSend is always c_void_p; but the
+            # argument types vary with the specific call.
+            objc.objc_msgSend.restype = c_void_p
+        else:
+            # Failed to load the objc library
+            objc = None
+
+    class IOSBrowser(BaseBrowser):
+        def open(self, url, new=0, autoraise=True):
+            sys.audit("webbrowser.open", url)
+            # If ctypes isn't available, we can't open a browser
+            if objc is None:
+                return False
+
+            # This is the equivalent of:
+            #    NSString url_string =
+            #        [NSString stringWithCString:url.encode("utf-8")
+            #                           encoding:NSUTF8StringEncoding];
+            NSString = objc.objc_getClass(b"NSString")
```

Comment:
`snake_case` for this and the other variables here, or is it more important to mirror the `objc` names?

Reply:
As with the `_ios_ver()` implementation, this is a "weak Hungarian" signifier of their ObjC origin.